AppController/djinn.rb

```diff
@@ -5576,21 +5573,26 @@ def fetch_revision(revision_key)
 
     RETRIES.downto(0) { ||
       begin
-        remote_machine = ZKInterface.get_revision_hosters(
+        ip = ZKInterface.get_revision_hosters(
           revision_key, @options['keyname']).sample
       rescue FailedZooKeeperOperationException
         sleep(SMALL_WAIT)
         next
       end
 
-      if remote_machine.nil?
+      if ip.nil?
         Djinn.log_info("Waiting for a machine to have a copy of #{app_path}")
         Kernel.sleep(SMALL_WAIT)
         next
       end
 
-      ssh_key = remote_machine.ssh_key
-      ip = remote_machine.private_ip
+      # Get the ssh key to use for the remote machine.
+      ssh_key = @nodes.find { |node| node["private_ip"] == ip }["ssh_key"]
+      if ssh_key.empty?
```

Comment:
`ssh_key` will be `nil` if the IP address is not found. Also, trying to get `["ssh_key"]` from `nil` will raise an exception.

Reply:
Good comments: thanks!